File: ParcelableListExample/app/src/main/java/com/example/velm/parcelablelistexample/MainActivity.java
package com.example.velm.parcelablelistexample;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    public void sendIntent(View view){
        List<User> userList = new ArrayList<>();
        User user1 = new User("User 1",22,"Chennai");
        userList.add(user1);
        User user2 = new User("User 2",25,"Banglore");
        userList.add(user2);

        Intent userListIntent = new Intent(getApplicationContext(),UserListActivity.class);
        userListIntent.putParcelableArrayListExtra("userlist",(ArrayList<? extends Parcelable>) userList);
        startActivity(userListIntent);
    }
}

File: BackgroundServiceTest/app/src/main/java/com/velmurugan/backgroundservicetest/MainActivity.kt
package com.velmurugan.backgroundservicetest

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    val MY_PERMISSIONS_REQUEST_LOCATION = 100

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED
        ) {
            if (ActivityCompat.shouldShowRequestPermissionRationale(this as Activity, Manifest.permission.ACCESS_FINE_LOCATION) || ActivityCompat.shouldShowRequestPermissionRationale(this as Activity, Manifest.permission.ACCESS_COARSE_LOCATION)) {
                // permission is denined by user, you can show your alert dialog here to send user to App settings to enable permission
            } else {
                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION), MY_PERMISSIONS_REQUEST_LOCATION)
            }
        }


        start.setOnClickListener {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                ContextCompat.startForegroundService(this@MainActivity, Intent(this@MainActivity, MyService::class.java))
            } else {
                startService(Intent(this@MainActivity, MyService::class.java))
            }
        }
    }
}